src/libs/Trie/TrieNode.js

```diff
@@ -0,0 +1,46 @@
+import _ from 'underscore';
+
+class TrieNode {
+    /**
+     * Reset all attributes to default values.
+     */
+    constructor() {
+        this.children = {};
+        this.leaf = false;
+        this.metaData = {};
+    }
+
+    /**
+     * Make the current as a leaf node.
+    */
+    setCompleteWord() {
+        this.leaf = true;
+    }
+
+    /**
+    * Check if the current node is a leaf node.
+    * @returns {Boolean}
+    */
+    isCompleteWord() {
+        return _.keys(this.children).length === 0;
+    }
```

Comment:
@Karim-30 I'm a little confused here, can you elaborate on the comment for `isCompleteWord`?

Reply:
@Santhosh-Sellavel nodes that don't have children called leaf nodes as @roryabraham explained [here](https://github.com/Expensify/App/pull/10165#discussion_r940447915) 

for example in the following tree `p` and `t` are leaf nodes.

        c
        |
        a
      /   \
     p     t